src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -593,15 +595,73 @@ public async Task<PlatformSummary> GetLatestElasticBeanstalkPlatformArn(Beanstal
+        /// For Linux beanstalk platforms the describe calls return a collection of .NET x and .NET Core based platforms.
+        /// The order returned will be sorted by .NET version in increasing order then by platform versions. So for example we could get a result like the following
+        ///
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.1.3
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.1.2
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.0.6
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.0.5
+        /// .NET 8 running on 64bit Amazon Linux 2023 v3.1.3
+        /// .NET Core running on 64bit Amazon Linux 2 v2.8.0
+        /// .NET Core running on 64bit Amazon Linux 2 v2.7.3
+        /// .NET Core running on 64bit Amazon Linux 2 v2.6.0
+        ///
+        /// We want the user to use the latest version of each platform first as well as the latest version of .NET first. Also .NET x should come before .NET Core.
+        /// The above example will be sorted into the following.
+        ///
+        /// .NET 8 running on 64bit Amazon Linux 2023 v3.1.3
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.1.3
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.1.2
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.0.6
+        /// .NET 6 running on 64bit Amazon Linux 2023 v3.0.5
+        /// .NET Core running on 64bit Amazon Linux 2 v2.8.0
+        /// .NET Core running on 64bit Amazon Linux 2 v2.7.3
+        /// .NET Core running on 64bit Amazon Linux 2 v2.6.0
+        /// </summary>
+        /// <param name="platforms"></param>
+        public static List<PlatformSummary> SortElasticBeanstalkLinuxPlatforms(List<PlatformSummary> platforms)
+        {
+            var dotnetVersionMap = new Dictionary<string, int>();
+            foreach (var platform in platforms)
+            {
+                var runningIndexOf = platform.PlatformBranchName.IndexOf("running", StringComparison.InvariantCultureIgnoreCase);
+                if (runningIndexOf == -1)
+                {
+                    dotnetVersionMap[platform.PlatformArn] = 0;
+                    continue;
+                }
+
+                var framework = platform.PlatformBranchName.Substring(0, runningIndexOf).Trim();
+                var frameworkSplit = framework.Split(" ");
+                if (frameworkSplit.Length != 2)
+                {
+                    dotnetVersionMap[platform.PlatformArn] = 0;
+                    continue;
+                }
+
+                if (!int.TryParse(frameworkSplit[1], out var dotnetVersion))
+                {
+                    dotnetVersionMap[platform.PlatformArn] = 0;
+                    continue;
+                }
+
+                dotnetVersionMap[platform.PlatformArn] = dotnetVersion;
+            }
 
+            return platforms.OrderByDescending(x => new Version(x.PlatformVersion)).ThenByDescending(x => dotnetVersionMap[x.PlatformArn]).ToList();
```

Comment:
I think this condition should be inverted. We first want to sort the platforms in descending order of the .NET version and then in descending order of platform versions. 
```suggestion
            return platforms.OrderByDescending(x => dotnetVersionMap[x.PlatformArn]).ThenByDescending(x => new Version(x.PlatformVersion)).ToList();
```